### DISCUSSION TITLE
How to change the Default data directory of the Nextcloud Digitalocean 1-click Droplet?

### DISCUSSION THREAD

**Discussion by S0ulf3re** [nextcloud/all-in-one#644] [Questions] [answered]
For context, I'm using the one at [https://marketplace.digitalocean.com/apps/nextcloud](url)

The documentation says to change the master container environment variable, but I'm not sure how to do that. Ive tried using "env export" but that doesn't seem to work as it gets forgotten.

**Comment by szaimen** [comment, marked as answer]
Hello, the Nextcloud datadir variable is on digital ocean by default set to `/mnt/ncdata`. See https://github.com/nextcloud-releases/all-in-one/blob/16e1c468317481128010badeabc5a342988c6f72/.build/digitalocean/files/var/lib/cloud/scripts/per-instance/001_onboot#L19

So if you mount your external drive there before the first startup (and adjust the permissions correctly), you do not need to adjust the container startup command...

**Reply by S0ulf3re** [reply]
So just create a droplet and then attach a volume with the name "ncdata"?

It seems that you have to attach the volume after the droplet boots because DO doesn't allow you to name volumes that you create with a droplet.

**Reply by szaimen** [reply]
> So just create a volume with the name "ncdata"?

I suppose so... as I said it needs to be mounted to `/mnt/ncdata` before starting Nextcloud the first time. (also please do not forget the apply the correct permissions. How to apply the correct permissions is documented here: 
![image](https://user-images.githubusercontent.com/42591237/168568838-677e290f-c770-48eb-acf2-f9b1ab5869a3.png)

> It seems that you have to attach the volume after the droplet boots because DO doesn't allow you to name volumes that you create with a droplet.

This is not a problem because the installation of Nextcloud is not started directly. You need to log in to the AIO interface first and follow the setup process in order to install Nextcloud. So before doing that you have time to adjust your external drive first...

**Reply by S0ulf3re** [reply]
ok
